crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -435,7 +437,10 @@ impl FromWorld for MeshPipeline {
                 TextureFormat::bevy_default(),
             );
             let texture = render_device.create_texture(&image.texture_descriptor);
-            let sampler = render_device.create_sampler(&image.sampler_descriptor);
+            let sampler = render_device.create_sampler(match image.sampler_descriptor {
```

Comment:
I wonder if it's worth caching the default sampler and only creating one of them instead of many samplers with the same sampler_descriptor. I've got some idea in general that we don't need as many samplers as we create and bind as most are going to be exactly the same. I don't feel that this has to be blocking this PR though, but I'd kind of like @cart to comment. In my mind, this is not any kind of regression, just that touching this code highlights that we create many samplers with the same descriptor.

Reply:
Ha, my memory these days... I'm repeating my comment from above: https://github.com/bevyengine/bevy/pull/4465#issuecomment-1099101285

Reply:
Hmm yeah that seems like it would be a pretty easy (and in some cases big) win. If @aevyrie has time (and motivation) for it, im down to wait. But if not, I don't think we need to block on it.

Reply:
I can take care of this, if I can get some assistance from @superdump :slightly_smiling_face:.

1. Where would we locate this cache? A resource?
2. How would we key into this, would we need to hash the descriptor, and generate new ones only when they miss the cache?
3. Would we be store the wgpu `Sampler` in the cache?

Reply:
1. I was thinking a resource in the render world.
2. For now the simplest thing to do would be to only do it for the default sampler. A generic sampler cache like the texture cache is beyond the scope of this PR I think.
3. Yes. I guess an `Option<Sampler>` if Sampler can be cheaply cloned, or some reference county thing like Arc(? I so rarely use these kinds of ideas anymore in Rust that I forget how they work within the confines of Rust where lifetimes and mutability are strictly enforced) if not.

Reply:
Okay. I've implemented default sampler caching. 

- As Rob requested on discord, I tested assigning the same sampler to 2 bind groups, and I did not get an error. 
- I also instrumented the caching function to report cache hits, and this seemed to be working as well.

Please take a close look to ensure I've done this correctly.

Reply:
In the short term I'm cool with treating this as static config as changing this at runtime seems like an unlikely scenario, and the implementation cost of doing that seems high.